helix-view/src/document.rs

```diff
@@ -587,9 +620,14 @@ impl Document {
         loader: Option<Arc<helix_core::syntax::Loader>>,
     ) {
         if let (Some(language_config), Some(loader)) = (language_config, loader) {
-            if let Some(highlight_config) = language_config.highlight_config(&loader.scopes()) {
-                let syntax = Syntax::new(&self.text, highlight_config, loader);
-                self.syntax = Some(syntax);
+            if self.highlight_syntax {
+                if let Some(highlight_config) = language_config.highlight_config(&loader.scopes()) {
+                    let syntax = Syntax::new(&self.text, highlight_config, loader);
+                    self.syntax = Some(syntax);
+                }
+            } else {
+                // If there is some syntax and it should be disabled, disable it.
+                self.syntax = None;
```

Comment:
If I'm understanding this right, this isn't really just disabling highlighting, it's disabling any and all syntax tree parsing. Is the issue of slow speed specifically due to highlight rendering, or is it caused by having such a large syntax tree loaded into memory at once? Disabling the syntax parsing still also disable text objects, auto complete, etc. 

If it's specifically caused by highlighting, then we may want to disable highlighting without disabling all of the syntax tree parsing. And if we do want to disable syntax parsing altogether, then I think we should rename the command and struct fields something like `enable_tree_sitter` or something.

Reply:
Yeah, you are completely right, this is more about disabling syntax parsing then highlighting. Highlighting (or rendering) by itself is not an issue, the parser is. I have experimented with the parser, but I have not found any better solution than disabling it completely.

I will rename commands and struct fields to names that are more accurate, as you have suggested.